packages/salesforcedx-vscode-soql/src/editor/soqlEditorProvider.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+export class SOQLEditorProvider implements vscode.CustomTextEditorProvider {
+  public static register(context: vscode.ExtensionContext) {
+    const provider = new SOQLEditorProvider(context);
+    const providerRegistration = vscode.window.registerCustomEditorProvider(
+      SOQLEditorProvider.viewType,
+      provider
+    );
+    return providerRegistration;
+  }
+
+  private static readonly viewType = 'soqlCustom.soql';
+  constructor(private readonly context: vscode.ExtensionContext) {}
+
+  public async resolveCustomTextEditor(
+    document: vscode.TextDocument,
+    webviewPanel: vscode.WebviewPanel,
+    _token: vscode.CancellationToken
+  ): Promise<void> {
+    webviewPanel.webview.options = {
+      enableScripts: true,
+      localResourceRoots: [
+        vscode.Uri.file(
+          path.join(
+            this.context.extensionPath,
+            'node_modules/@salesforce/soql-builder-ui'
+          )
+        )
+      ]
+    };
+
+    webviewPanel.webview.html = this.getWebViewContent(webviewPanel.webview);
```

Comment:
It's a best practice to have a content security policy in webviews.
See: https://github.com/forcedotcom/soql-tooling/pull/2#discussion_r462531149

Reply:
This has been addressed in the other PR [here](https://github.com/forcedotcom/soql-tooling/pull/2#discussion_r462606570)